From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Wed, 30 Nov 2022 20:17:41 +0100
Subject: [PATCH] Check frozen ticks before landing block

License: LGPL-3.0-only (https://www.gnu.org/licenses/lgpl-3.0.html)
Gale - https://galemc.org

This patch is based on the following mixin:
"net/caffeinemc/mods/lithium/mixin/entity/fast_powder_snow_check/LivingEntityMixin.java"
By: 2No2Name <2No2Name@web.de>
As part of: Lithium (https://github.com/CaffeineMC/lithium)
Licensed under: LGPL-3.0-only (https://www.gnu.org/licenses/lgpl-3.0.html)

// SYNC SOURCE START
rev: 01597fb879b7a6d17c2420ec4e25a98df3bc8e7b
branch: develop
repo: https://github.com/CaffeineMC/lithium.git
track:
- "common/src/main/java/net/caffeinemc/mods/lithium/mixin/entity/fast_powder_snow_check/LivingEntityMixin.java"
// SYNC SOURCE END

diff --git a/net/minecraft/world/entity/LivingEntity.java b/net/minecraft/world/entity/LivingEntity.java
index a9eca4018a31dddeae4da85088097eb49e45ca06..3f554b998df6811f81e6825f274775765e9aab42 100644
--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -576,10 +576,9 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
     }
 
     protected void tryAddFrost() {
-        if (!this.getBlockStateOnLegacy().isAir()) {
             int ticksFrozen = this.getTicksFrozen();
             if (ticksFrozen > 0) {
-                AttributeInstance speed = this.getAttribute(Attributes.MOVEMENT_SPEED);
+                AttributeInstance speed = this.getBlockStateOnLegacy().isAir() ? null : this.getAttribute(Attributes.MOVEMENT_SPEED); // Gale - Lithium - check frozen ticks before landing block
                 if (speed == null) {
                     return;
                 }
@@ -587,7 +586,6 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
                 float slowAmount = -0.05F * this.getPercentFrozen();
                 speed.addTransientModifier(new AttributeModifier(SPEED_MODIFIER_POWDER_SNOW_ID, slowAmount, AttributeModifier.Operation.ADD_VALUE));
             }
-        }
     }
 
     protected void onChangedBlock(final ServerLevel level, final BlockPos pos) {
